packages/controllers/src/snaps/RequestQueue.ts

```diff
@@ -0,0 +1,47 @@
+export class RequestQueue {
+  public readonly maxQueue: number;
+
+  private readonly locks: Map<string, number>;
+
+  constructor(maxQueue: number) {
+    this.maxQueue = maxQueue;
+    this.locks = new Map<string, number>();
+  }
+
+  /**
+   * Increments the queue count for a particular origin
+   *
+   * @param origin
+   */
+  public increment(origin: string) {
+    const currentCount = this.locks.get(origin) ?? 0;
+    if (currentCount + 1 > this.maxQueue) {
```

Comment:
Perhaps can be simplified to `currentCount >= this.maxQueue`?

Reply:
true